Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1

```diff
@@ -0,0 +1,1159 @@
+    }
+
+    $currentProperty += "$indent}"
+    #if ($isArray -or $IndentLevel -gt 4)
+    #{
+        $currentProperty += "`r`n"
+    #}
+
+    #Indenting last parenthese when the cim instance is an array
+    if ($IndentLevel -eq 5)
+    {
+        $indent = '    ' * ($IndentLevel -2)
+        $currentProperty += $indent
+    }
+
+    $emptyCIM = $currentProperty.replace(' ', '').replace("`r`n", '')
+    if ($emptyCIM -eq "MSFT_$CIMInstanceName{}")
+    {
+        $currentProperty = $null
+    }
+    return $currentProperty
+}
+
+Function Get-M365DSCDRGSimpleObjectTypeToString
+{
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param(
+        [Parameter(Mandatory = 'true')]
+        [System.String]
+        $Key,
+
+        [Parameter(Mandatory = 'true')]
+        $Value,
+
+        [Parameter()]
+        [System.String]
+        $Space = '                '
+
+    )
+
+    $returnValue = ''
+    switch -Wildcard ($Value.GetType().Fullname )
+    {
+        '*.Boolean'
+        {
+            $returnValue = $Space + $Key + " = `$" + $Value.ToString() + "`r`n"
+        }
+        '*.String'
+        {
+            if ($key -eq '@odata.type')
+            {
+                $key = 'odataType'
+            }
+            $returnValue = $Space + $Key + " = '" + $Value + "'`r`n"
+        }
+        '*.DateTime'
+        {
+            $returnValue = $Space + $Key + " = '" + $Value + "'`r`n"
+        }
+        '*[[\]]'
+        {
+            $returnValue = $Space + $key + ' = @('
+            $whitespace = ''
+            $newline = ''
+            if ($Value.count -gt 1)
+            {
+                $returnValue += "`r`n"
+                $whitespace = $Space + '    '
+                $newline = "`r`n"
+            }
+            foreach ($item in ($Value | Where-Object -FilterScript { $null -ne $_ }))
+            {
+                switch -Wildcard ($item.GetType().Fullname)
+                {
+                    '*.String'
+                    {
+                        $returnValue += "$whitespace'$item'$newline"
+                    }
+                    '*.DateTime'
+                    {
+                        $returnValue += "$whitespace'$item'$newline"
+                    }
+                    Default
+                    {
+                        $returnValue += "$whitespace$item$newline"
+                    }
+                }
+            }
+            if ($Value.count -gt 1)
+            {
+                $returnValue += "$Space)`r`n"
+            }
+            else
+            {
+                $returnValue += ")`r`n"
+
+            }
+        }
+        Default
+        {
+            $returnValue = $Space + $Key + ' = ' + $Value + "`r`n"
+        }
+    }
+    return $returnValue
+}
+
+function Compare-M365DSCComplexObject
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param(
+        [Parameter()]
+        $Source,
+        [Parameter()]
+        $Target
```

Comment:
Does this parameter have a specific type? e.g. [Object] - Which would not be necessary, but would make it a bit clearer what this function is capable of. see> https://github.com/PowerShell/PowerShell/issues/12359